client/js/components/tags/TagsList.vue

```diff
@@ -0,0 +1,487 @@
+import TagListHeader from './TagListHeader'
+
+
+export default {
+  name: 'TagsList',
+
+  components: {
+    AddonWrapper,
+    DpButton,
+    DpCheckbox,
+    DpIcon,
+    DpInput,
+    DpLoading,
+    DpModal,
+    DpUpload,
+    DpSelect,
+    DpTreeList,
+    TagsImportForm,
+    TagListBulkControls,
+    TagListEditForm,
+    TagListHeader
+  },
+
+  props: {
+    procedureId: {
+      type: String,
+      required: true
+    }
+  },
+
+  mixins: [dpValidateMixin],
+
+  data() {
+    return {
+      dataIsRequested: false,
+      isInEditState: '',
+      newTag: {
+        title: '',
+        topic: ''
+      },
+      newTopic: {
+        title: ''
+      },
+      // This is necessary to allow extending the Tags-Resource
+      tagAttributes: {
+        boilerplate: '',
+        title: ''
+      }
+    }
+  },
+
+  computed: {
+    ...mapState('Tag', {
+      Tag: 'items'
+    }),
+    ...mapState('TagTopic', {
+      TagTopic: 'items'
+    }),
+
+    tagAttributeKeys () {
+      return Object.keys(this.tagAttributes)
+    },
+
+    topicsAsOptions () {
+      return Object.values(this.TagTopic).map(category => {
+        const { attributes, id } = category
+
+        return {
+          label: attributes.title,
+          value: id
+        }
+      })
+    },
+
+    transformedCategories () {
+      return Object.values(this.TagTopic).map(category => {
+        const { attributes, id, type } = category
+        const tags = category.relationships?.tags?.data.length > 0 ? category.relationships.tags.list() : []
+
+        return {
+          id,
+          attributes,
+          children: Object.values(tags).map(tag => {
+            const { attributes, id, relationships, type } = tag
+            const boilerplate = relationships?.boilerplate?.get ? relationships.boilerplate.get() : null
+
+            return {
+              attributes,
+              id,
+              relationships: { boilerplate },
+              type
+            }
+          }),
+          type
+        }
+      })
+    }
+  },
+
+  methods: {
+    ...mapMutations('Tag', {
+      updateTag: 'setItem'
+    }),
+
+    ...mapMutations('TagTopic', {
+      updateTagTopic: 'setItem'
+    }),
+
+    ...mapActions('Tag', {
+      createTag: 'create',
+      listTags: 'list',
+      saveTag: 'save'
+    }),
+
+    ...mapActions('TagTopic', {
+      createTagTopic: 'create',
+      listTagTopics: 'list',
+      saveTagTopic: 'save'
+    }),
+
+    abort () {
+      this.isInEditState = ''
+    },
+
+    branchFunc () {
+      return function ({ node }) {
+        return node.type === 'TagTopic'
+      }
+    },
+
+    changeTopic ({ elementId, parentId }) {
+      console.log('changeTopic', { elementId, parentId })
+
+      if (!parentId) {
+        console.error('No parentId provided:', { elementId, parentId })
+
+        return
+      }
+
+      const newParent = !this.TagTopic[parentId].relationships.tags.data.find(tag => tag.id === elementId)
```

Comment:
if it is a boolean value, it could be named smth like `isNewParent` for the better readability